File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/message/ReadChildAttributeRequest.java
package org.opcfoundation.webserver.types.message;

import org.jspecify.annotations.Nullable;
import org.opcfoundation.webserver.addressspace.nodes.UaVariableType;
import org.opcfoundation.webserver.types.UaChildId;
import org.opcfoundation.webserver.types.UaObjectId;

public class ReadChildAttributeRequest {
    private final UaObjectId objectId;
    private final UaChildId childId;
    private final @Nullable UaVariableType childVariableType;

    public ReadChildAttributeRequest(
            UaObjectId objectId,
            UaChildId childId,
            @Nullable UaVariableType childVariableType)
    {
        this.objectId = objectId;
        this.childId = childId;
        this.childVariableType = childVariableType;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public UaChildId getChildId() {
        return childId;
    }

    public @Nullable UaVariableType getChildVariableType() {
        return childVariableType;
    }
}

File: opcua-webservice-node/src/server/service/message/reactiveobject/ReadChildAttributeRequest.ts
import { UaVariableType } from "../../../addressspace/nodes/UaVariableType";
import { UaChildId, UaObjectId } from "../../../types";

export class ReadChildAttributeRequest {
    private readonly _objectId: UaObjectId;
    private readonly _childId: UaChildId;
    private readonly _childVariableType: UaVariableType | null;

    constructor(
        objectId: UaObjectId,
        childId: UaChildId,
        childVariableType: UaVariableType | null,
    ) {
        this._objectId = objectId;
        this._childId = childId;
        this._childVariableType = childVariableType;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    get childId(): UaChildId {
        return this._childId;
    }

    get childVariableType(): UaVariableType | null {
        return this._childVariableType;
    }
}